### DISCUSSION TITLE
How to test events sent via Alpine to Livewire component?

### DISCUSSION THREAD

**Discussion by Kovah** [livewire/livewire#7486] [1. Help] [answered]
I have a Livewire component which listens to an event which is only fired from a button:

```php
<?php
class Wrapper extends Component
{
    #[On('trigger-new-site')]
    public function getSite(): void
    {
        // do stuff here
    }
}
```
```html
<button wire:click="$dispatch('trigger-new-site')">
  Next Site
</button>
```

The question is: how do I automatically test this in PHPunit?

I already tried dispatching the event from another component like this:
```php
$wrapper = Livewire::test(Wrapper::class)
    ->assertSet('site', $site);

Livewire::test(Frame::class)->dispatch('trigger-new-site');

$wrapper->assertSet('site', $site2);
```
However, I am getting a `EventHandlerDoesNotExist` exception. Is there any way to _globally_ dispatch a Livewire event and then check if the component listened to it? Unfortunately, this use case is not covered in [the docs](https://livewire.laravel.com/docs/testing#events).

**Comment by joshhanley** [comment]
@Kovah Livewire tests are run using PHPUnit, and are not in the browser. So you can't test the interactivity between Livewire and Alpine as Alpine only runs in the browser. If you want to test that, you could use Laravel Dusk to run some browser tests.

Hope this helps!

**Reply by Kovah** [reply]
Hi, yes I know that Alpine is a frontend tool.
I have that Livewire component listening to an event, and it seems that there is no way to test this listener if I have no other component which dispatches the event. Should I just run `->call('getSite')` on the Wrapper component and hope that the event listening is working correctly? I would more think of something like `Livewire::dispatchEvent('some-event', $params)`. 🤔

**Reply by joshhanley** [reply, marked as answer]
@Kovah oh sorry, I misunderstood. 

Yeah you can call dispatch on the wrapper component to make sure it reacts to the event correctly, like

```php
Livewire::test(Wrapper::class)
    ->assertSet('site', $site)
    ->dispatch('trigger-new-site')
    ->assertSet('site', $site2);
```

See here for details https://livewire.laravel.com/docs/events#testing-event-listeners

Hope this helps!

**Reply by Kovah** [reply]
Ah thanks! Seems I misread the docs at this point.

**Reply by joshhanley** [reply]
No worries! 🙂
